### PR TITLE
[geoip-lite] add more defenitions; fix some types

### PR DESCRIPTION
**Pull request by witem**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/geoip-lite/node-geoip/blob/v1.4.2/lib/geoip.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## geoip-lite ([<kbd>unpkg</kbd>](https://unpkg.com/browse/geoip-lite@latest/))
was missing the following properties:
1. cmp
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5d9e9905238c51d151960b35958552cef742697f
</p>

**Comment by witem** [comment]
Hi 
@uniqueiniquity or @sandersn maybe you can review this changes? 
Thanks, also I want to contribute this package, what changes I need to make?

File: types/geoip-lite/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for GeoIP-lite 1.1.6
+// Type definitions for GeoIP-lite 1.4.2
 // Project: https://github.com/bluesmoon/node-geoip
 // Definitions by: Yuce Tekol <http://yuce.me/>
```
Comment (sandersn) [inline types/geoip-lite/index.d.ts:3]: @witem add yourself here to be counted as a package owner in future PRs

**Comment by sandersn** [comment]
@yuce can you take a look at this?

File: types/geoip-lite/index.d.ts (hunk 1 above)
Comment (witem) [reply types/geoip-lite/index.d.ts:3]: thanks, I added

**Comment by witem** [comment]
Hi @sandersn can you review code again?

File: types/geoip-lite/index.d.ts [hunk 2]
```diff
@@ -33,9 +34,20 @@ declare namespace mod {
...
+    interface AsyncCallback {
+        (err?: Error): void;
+    }
+
+    export type CmpArgs = number | [number];
+    export type CmpResult = 1 | -1 | 0 | null;
+
+    export function cmp(a: CmpArgs, b: CmpArgs): null | Lookup;
+    export function lookup(ip: string | number): null | Lookup;
+    export function pretty(ip: string | number | [string | number]): string;
```
Comment (sandersn) [inline types/geoip-lite/index.d.ts:46]:
```suggestion
    export function pretty(ip: string | number | Array<string | number>): string;
```

(`[string | number]` is a singleton of string or number, not an array of string or number)

**Review by sandersn — CHANGES_REQUESTED** [review]
I don't know anything about geoip-lite, so I'd like @yuce to take a look too.

File: types/geoip-lite/index.d.ts (hunk 2 above)
Comment (witem) [reply types/geoip-lite/index.d.ts]: yes, thanks

**Comment by witem** [comment]
Hi @sandersn , seems @yuce not come. Can you review it again and approve this PR?

**Comment by witem** [comment]
@uniqueiniquity or @sandersn can anyone approve this PR? I'm wait more that 8 days after last change. Thanks!

**Review by sandersn — APPROVED** [review]

**Comment by sandersn** [comment]
Thanks for your patience on this one @witem.

**Pull request merged**